src/response_sender.cc

```diff
@@ -123,6 +126,9 @@ ResponseSender::Send(
 
   CheckResponseSenderArguments(infer_response, flags);
   UpdateStateAndCounters(infer_response, flags);
+  if (infer_response) {
+    infer_response->PruneOutputTensors(requested_output_names_);
```

Comment:
`requested_output_names_` is guaranteed to be full set of outputs if not specified by the client?

Reply:
Yes, because underneath it is calling [this two C APIs](https://github.com/triton-inference-server/core/blob/main/src/backend_model.cc#L1292-L1333) and the `ImmutableRequestedOutputs()` used by both will check if `requested_outputs_` is empty and use `original_requested_outputs_` if so.